source/frontend/src/components/maps/leaflet/Layers/PointClusterer.tsx

```diff
@@ -258,148 +215,142 @@ export const PointClusterer: React.FC<React.PropsWithChildren<PointClustererProp
-                    });
-                  }
-                },
-              }}
+              pointFeature={clusterFeature}
+              markerPosition={m.position}
+              isSelected={false}
+              isOwned={isOwned}
             />
           );
         })}
         {/**
          * Render lines/legs from a spiderfied cluster click
          */}
-        {spider.lines?.map((m: any, index: number) => (
+        {spider.lines?.map((m, index: number) => (
           <Polyline key={index} positions={m.coords} {...m.options} />
         ))}
-        {/**
-         * render selected property marker, auto opens the property popup
-         */}
-        {!!selected &&
-          selected?.id === currentSelected?.id &&
-          !currentClusterIds.includes(Number(selected?.id)) && (
-            <SelectedPropertyMarker
-              {...selected}
-              icon={getMarkerIcon({ properties: selected } as any, true)}
-              className={'active-selected'}
-              position={[selected.latitude as number, selected.longitude as number]}
-              map={mapInstance}
-              eventHandlers={{
-                click: () => {
-                  mapStateContext.setState({
-                    type: MapStateActionTypes.SELECTED_INVENTORY_PROPERTY,
-                    selectedInventoryProperty: selected,
-                  });
-                  selected && onMarkerClick(selected);
-                },
-              }}
-            />
-          )}
       </FeatureGroup>
+      {/**
+       * Render all of the unclustered DRAFT MARKERS.
+       **/}
       <FeatureGroup ref={draftFeatureGroupRef}>
         {draftPoints.map((draftPoint, index) => {
-          //render all of the unclustered draft markers.
-          if (draftPoint?.geometry?.coordinates) {
-            const [longitude, latitude] = draftPoint.geometry.coordinates;
-            if (!latitude || !longitude) return <></>;
-            return (
-              <Marker
-                {...(draftPoint.properties as any)}
-                key={index}
-                position={[latitude, longitude]}
-                icon={getDraftIcon((index + 1).toString())}
-                zIndexOffset={500}
-              ></Marker>
-            );
-          }
-          return <></>;
+          return (
+            <Marker
+              key={index}
+              position={draftPoint}
+              icon={getDraftIcon((index + 1).toString())}
+              zIndexOffset={500}
+            ></Marker>
+          );
         })}
       </FeatureGroup>
     </>
   );
 };
 
 export default PointClusterer;
+
+/**
+ * @param feature the feature to obtain lat/lng coordinates for.
+ * @returns [lat, lng]
+ */
+const getLatLng = <P,>(feature: Feature<Geometry, P>) => {
+  if (feature.geometry.type === 'Polygon' || feature.geometry.type === 'MultiPolygon') {
+    const latLng = geoJSON(feature.geometry).getBounds().getCenter();
```

Comment:
I think getCenter is bad times because it can't guarantee that the returned lat/lng is actually within the boundary (think of a property with a boundary shaped like the letter C). I think we want to user polylabel for this.

Reply:
That would be a functional change. Lets create a task for that